src/etcd/client.py

```diff
@@ -223,8 +226,11 @@ def _set_version_info(self):
         self._version = version_info['etcdserver']
         self._cluster_version = version_info['etcdcluster']
 
-    def _discover(self, domain):
-        srv_name = "_etcd._tcp.{}".format(domain)
+    def _discover(self, domain, use_ssl=False):
```

Comment:
I don't like this change as it breaks compatibility with the previous version.

I would make discovery first search for the new alias, then fallback to the old one if discovery returns nothing.

Historically we had client SRV support before the official client, and we went with the same key that confd is using. I still see a lot of people might want to preserve that old name.
It should be fairly easy to implement too.

Also, since we save protocol in self.protocol, use that instead of adding a parameter.